File: mobile/android/app/src/main/java/com/cdbl/leavemanager/data/api/DashboardService.kt
package com.cdbl.leavemanager.data.api

import com.cdbl.leavemanager.data.model.BalanceResponse
import com.cdbl.leavemanager.data.model.BalanceDetailResponse
import com.cdbl.leavemanager.data.model.TeamOnLeaveResponse
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.Header
import retrofit2.http.POST
import retrofit2.http.Body
import retrofit2.http.PATCH
import retrofit2.http.Path

interface DashboardService {
    @GET("balance/mine")
    suspend fun getMyBalance(@Header("Authorization") token: String): Response<BalanceResponse>

    @GET("balance/mine")
    suspend fun getMyBalanceDetailed(
        @Header("Authorization") token: String,
        @retrofit2.http.Query("detailed") detailed: Boolean = true
    ): Response<BalanceDetailResponse>

    @GET("team/on-leave?scope=team")
    suspend fun getTeamOnLeave(@Header("Authorization") token: String): Response<TeamOnLeaveResponse>

    @GET("admin/system-stats")
    suspend fun getSystemStats(@Header("Authorization") token: String): Response<com.cdbl.leavemanager.data.model.SystemStatsResponse>

    @GET("admin/logs")
    suspend fun getAuditLogs(@Header("Authorization") token: String): Response<com.cdbl.leavemanager.data.model.AuditLogsResponse>

    @GET("dashboard/hr/stats")
    suspend fun getHRStats(@Header("Authorization") token: String): Response<com.cdbl.leavemanager.data.model.HRAdminStats>

    @GET("dashboard/ceo/stats")
    suspend fun getCEOStats(@Header("Authorization") token: String): Response<com.cdbl.leavemanager.data.model.CEOStats>

    @GET("dashboard/hr-head/stats")
    suspend fun getHRHeadStats(@Header("Authorization") token: String): Response<com.cdbl.leavemanager.data.model.HRHeadStats>

    @GET("admin/users")
    suspend fun getUsers(@Header("Authorization") token: String): Response<com.cdbl.leavemanager.data.model.AdminUserListResponse>

    @POST("admin/users/create")
    suspend fun createUser(
        @Header("Authorization") token: String,
        @Body request: com.cdbl.leavemanager.data.model.CreateUserRequest
    ): Response<com.cdbl.leavemanager.data.model.CreateUserResponse>

    @PATCH("admin/users/{id}")
    suspend fun updateUser(
        @Header("Authorization") token: String,
        @Path("id") id: String,
        @Body request: com.cdbl.leavemanager.data.model.UpdateUserRequest
    ): Response<com.cdbl.leavemanager.data.model.UpdateUserResponse>

    @GET("holidays")
    suspend fun getHolidays(@Header("Authorization") token: String): Response<com.cdbl.leavemanager.data.model.HolidayResponse>
}

File: mobile/ios/CDBLLeaveManager/CDBLLeaveManager/Core/Networking/DashboardService.swift
//
//  DashboardService.swift
//  CDBLLeaveManager
//
//  Dashboard statistics API service for all user roles.
//

import Foundation

actor DashboardService {
    static let shared = DashboardService()
    
    private let client = APIClient.shared
    
    private init() {}
    
    // MARK: - Employee Dashboard
    
    func getEmployeeDashboard() async throws -> EmployeeDashboardData {
        return try await client.request("dashboard/employee")
    }
    
    func getLeaveBalance() async throws -> DashboardLeaveBalance {
        // Correct endpoint: /api/balance/mine
        return try await client.request("balance/mine")
    }
    
    func getWhosOutToday() async throws -> [WhosOutMember] {
        let response: APIResponse<[WhosOutMember]> = try await client.request("dashboard/whos-out")
        return response.data ?? []
    }
    
    // MARK: - Manager Dashboard
    
    func getManagerStats() async throws -> ManagerStatsResponse {
        return try await client.request("dashboard/manager")
    }
    
    func getPendingApprovals() async throws -> ApprovalListResponse {
        return try await client.request("approvals/pending")
    }
    
    // MARK: - HR Admin Dashboard
    
    func getHRAdminStats() async throws -> HRAdminStats {
        // Correct endpoint: /api/dashboard/hr-admin/stats
        return try await client.request("dashboard/hr-admin/stats")
    }
    
    // MARK: - HR Head Dashboard
    
    func getHRHeadStats() async throws -> HRHeadStats {
        // Correct endpoint: /api/dashboard/hr-head/stats
        return try await client.request("dashboard/hr-head/stats")
    }
    
    // MARK: - CEO Dashboard
    
    func getCEOStats() async throws -> CEOStats {
        // Correct endpoint: /api/dashboard/ceo/stats
        return try await client.request("dashboard/ceo/stats")
    }
    
    // MARK: - System Admin Dashboard
    
    func getSystemStats() async throws -> SystemStatsResponse {
        return try await client.request("admin/system-stats")
    }
    
    func getRecentAuditLogs(limit: Int = 10) async throws -> AuditLogsResponse {
        // Correct endpoint: /api/admin/logs
        return try await client.request("admin/logs?limit=\(limit)")
    }
    
    // MARK: - Holidays
    
    func getHolidays(year: Int? = nil) async throws -> HolidayResponse {
        var endpoint = "holidays"
        if let year = year {
            endpoint += "?year=\(year)"
        }
        return try await client.request(endpoint)
    }
    
    // MARK: - Team Calendar
    
    func getTeamCalendar(month: Int, year: Int) async throws -> TeamCalendarResponse {
        return try await client.request(
            "dashboard/team-calendar?month=\(month)&year=\(year)"
        )
    }
}
